lib/simctl.js

```diff
@@ -650,8 +650,42 @@ async function getPasteboard (udid, encoding = 'utf-8') {
   }
 }
 
+/**
+ * Get the list of device types available in the current Xcode installation
+ *
+ * @return {Array<string>} List of the types of devices available
+ * @throws {Error} If the corresponding simctl command fails
+ */
+async function getDeviceTypes () {
+  try {
+    const {stdout} = await simExec('list', 0, ['devicetypes', '-j']);
+    /*
+     * JSON will be like:
+     *   {
+     *     "devicetypes" : [
+     *       {
+     *         "name" : "iPhone 4s",
+     *         "identifier" : "com.apple.CoreSimulator.SimDeviceType.iPhone-4s"
+     *       },
+     *       ...
+     *   }
+     */
+    const deviceTypes = JSON.parse(stdout);
+    return deviceTypes.devicetypes.map(function (type) {
+      return type.name;
+    });
+  } catch (err) {
+    let msg = `Unable to get list of device types: ${err.message}`;
+    if (err.stderr) {
+      msg = `${msg}. Stderr: ${err.stderr}`;
+    }
+    log.errorAndThrow(msg);
```

Comment:
I usually prefer not to use `errorAndThrow` in library functions. The client might deliberately decide to ignore the exception, but this call will anyway print it in the log with `error` priority, which might be confusing.